server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -853,4 +853,41 @@ public static String padStart(String s, int minimumLength, char c) {
             return sb.toString();
         }
     }
+
+    /**
+     * Checks that the specified String reference is:
+     *
+     * <ul>
+     * <li>not {@code null} and throws a customized {@link NullPointerException} if it is.
+     * <li>not empty (i.e. has zero length) and throws an  {@link IllegalArgumentException} if it is.
+     * </ul>
+     *
+     * This method is designed primarily for doing parameter validation in methods and
+     * constructors with multiple parameters, as demonstrated below:
+     *
+     * <blockquote><pre>
+     * public Foo(String bar, String baz) {
+     *     this.bar = Strings.requireNonEmpty(bar, "bar must not be null or empty");
+     *     this.baz = Strings.requireNonEmpty(baz, "baz must not be null or empty");
+     * }
+     * </pre></blockquote>
+     *
+     * @param obj     the String reference to check for nullity or zero-length
+     * @param message detail message to be used in the event that an exception is thrown
+     * @return {@code obj} if not {@code null} or zero-length
+     * @throws NullPointerException if {@code obj} is {@code null}
+     * @throws IllegalArgumentException if {@code obj.isEmpty()} returns {@code true}
+     */
+    public static String requireNonEmpty(String obj, String message) {
```

Comment:
Still outstanding -- let's remove this and just use `Objects.requireNonNull`.

Reply:
> an empty UUID string is kinda valid

I'm not sure I agree with this statement. The API is designed to import a specific UUID. You can't even POST to the REST API without passing a UUID, because the REST layer infers that you ought to sending a GET.

But, putting that aside, it's highly unlikely that anything is going to call the transport API but our code, so I've swapped the calls out as suggested.